pkg/apis/serving/v1beta1/service_validation.go

```diff
@@ -61,3 +63,24 @@ func (ss *ServiceStatus) Validate(ctx context.Context) *apis.FieldError {
 	return ss.ConfigurationStatusFields.Validate(ctx).Also(
 		ss.RouteStatusFields.Validate(ctx))
 }
+
+func (s *Service) validateCreatorAndModifier(ctx context.Context) *apis.FieldError {
+	if len(s.GetAnnotations()) == 0 {
+		return nil
+	}
+	var errs *apis.FieldError
+	if apis.IsInUpdate(ctx) {
+		original := apis.GetBaseline(ctx).(*Service)
+		if original.GetAnnotations()[serving.CreatorAnnotation] != s.GetAnnotations()[serving.CreatorAnnotation] {
+			errs = errs.Also(&apis.FieldError{
+				Message: "annotation " + serving.CreatorAnnotation + " is immutable",
```

Comment:
I'd use `fmt.Sprintf("annotation %q is immutable", serving.CreatorAnnotation)`.

Doesn't the `FieldError` type also take a path? If so, we should add the annotation there as well.